pkg/internal/cluster/providers/provider/common/proxy.go

```diff
@@ -34,12 +33,12 @@ const (
 
 // GetProxyEnvs returns a map of proxy environment variables to their values
 // If proxy settings are set, NO_PROXY is modified to include the cluster subnets
-func GetProxyEnvs(cfg *config.Cluster) map[string]string {
+func GetProxyEnvs(cfg *config.Cluster, env Getenver) map[string]string {
```

Comment:
the public API doesn't need changing, just make the public a wrapper around a private method that passes in the real getenv

Reply:
I implemented this in https://github.com/kubernetes-sigs/kind/pull/969